nebula-logger/core/main/logger-engine/lwc/logger/loggerService.js

```diff
@@ -66,71 +121,125 @@ const LoggerService = class {
-        ...settings,
-        supportedLoggingLevels: Object.freeze(settings.supportedLoggingLevels),
-        userLoggingLevel: Object.freeze(settings.userLoggingLevel)
-      });
-    } catch (error) {
-      /* eslint-disable-next-line no-console */
-      console.error(error);
-      throw error;
-    }
-  }
+    const loadSettingsTask = async () => {
+      try {
+        const retrievedSettings = await getSettings();
+        this.#settings = Object.freeze({
+          ...retrievedSettings,
+          supportedLoggingLevels: Object.freeze(retrievedSettings.supportedLoggingLevels),
+          userLoggingLevel: Object.freeze(retrievedSettings.userLoggingLevel)
+        });
+      } catch (error) {
+        /* eslint-disable-next-line no-console */
+        console.error(error);
+        throw error;
+      }
+    };
 
-  _meetsUserLoggingLevel(logEntryLoggingLevel) {
-    return this.#settings.isEnabled === true && this.#settings.userLoggingLevel.ordinal <= this.#settings?.supportedLoggingLevels[logEntryLoggingLevel];
+    this.#taskQueue.enqueueTask(loadSettingsTask);
+    await this.#taskQueue.executeTasks();
   }
 
   _newEntry(loggingLevel, message, originStackTraceError) {
     originStackTraceError = originStackTraceError ?? new Error();
-    const logEntryBuilder = newLogEntry(loggingLevel, this.#settings?.isConsoleLoggingEnabled, this.#settings?.isLightningLoggerEnabled)
+    const logEntryBuilder = new LogEntryEventBuilder(loggingLevel, new BrowserContext())
       .parseStackTrace(originStackTraceError)
       .setMessage(message)
       .setScenario(this.#scenario);
-    if (this._meetsUserLoggingLevel(loggingLevel)) {
-      this.#componentLogEntries.push(logEntryBuilder.getComponentLogEntry());
-    }
+    const logEntry = logEntryBuilder.getComponentLogEntry();
+
+    const loggingLevelCheckTask = providedLoggingLevel => {
+      if (this._meetsUserLoggingLevel(providedLoggingLevel)) {
+        this.#componentLogEntries.push(logEntry);
+
+        if (this.#settings.isConsoleLoggingEnabled) {
+          this._logToConsole(logEntry.loggingLevel, logEntry.message, logEntry);
+        }
+        if (this.#settings.isLightningLoggerEnabled) {
+          lightningLog(logEntry);
+        }
+      }
+    };
+
+    this.#taskQueue.enqueueTask(loggingLevelCheckTask, loggingLevel);
+    this.#taskQueue.executeTasks();
 
     return logEntryBuilder;
   }
-};
 
-const createLoggerService = async function () {
-  const service = new LoggerService();
-  await service._loadSettingsFromServer();
-  return service;
-};
+  _meetsUserLoggingLevel(logEntryLoggingLevel) {
+    return this.#settings.isEnabled === true && this.#settings.userLoggingLevel.ordinal <= this.#settings.supportedLoggingLevels[logEntryLoggingLevel];
+  }
 
-export { createLoggerService };
+  /* eslint-disable no-console */
+  _logToConsole(loggingLevel, message, componentLogEntry) {
+    let consoleLoggingFunction;
+    switch (loggingLevel) {
```

Comment:
what if instead of the switch statement here you used:

```js
let consoleLoggingFunction = console[loggingLevel.toLowerCase()];
if (!consoleLoggingFunction) {
  consoleLoggingFunction = console.debug;
}
```

Reply:
...but wait, what about this one-liner version??

```javascript
const consoleLoggingFunction = console[loggingLevel.toLowerCase()] ?? console.debug;
```

Reply:
![I've_Looked_At_This_For_Five_Hours_Now_Banner](https://github.com/user-attachments/assets/cf2c2b42-f203-41d7-bf19-7ccc995a1f7f)